redis/src/lib.rs

```
@@ -285,6 +285,24 @@
 //! # Ok(()) }
 //! ```
 //!
+//! If the user wants to handle server errors while also receiving values from successful requests,
+//! the user can define the target type of a pipeline query as `Vec<RedisResult<T>>` or `Vec<Value>`. This vector
+//! will contain only non-ignored values, if all requests completed successfully. If some failed, this
+//! vector will contain all server responses.
+//!
+//! ```rust,no_run
+//! # fn do_something() -> redis::RedisResult<()> {
+//! # let client = redis::Client::open("redis://127.0.0.1/").unwrap();
+//! # let mut con = client.get_connection().unwrap();
+//! let results : Vec<RedisResult<String>> = redis::pipe() //results: Vec<redis::Value> would also work
+//!         .set("x", "").ignore()
+//!         .hset("x", "field", "field_value").ignore()
+//!         .get("x")
+//!         .query(&mut con)?; // This will result with a vec![Ok("OK"), Err(WRONGTYPE), OK("x-value")]
```

For me this one feels wrong, probably you thinked about this but, before `Vec<RedisResult<T>>` you could get an error for failure of the whole pipe, now you will get everything even successful command results you have ignored before. 
Can you explain more about the approach you have took ?

---

It's a fair question. I think that some users might want both the successes and the failures when they happen, and I didn't want to add another function to opt into this behavior, I thought that using a type that represents an error is implicitly a way of showing you're interested in this.
Now, if you're interested in errors, we can't skip an error that would've previously failed the request - that would hide the error from the user. So we can't skip ignored failed responses.

If you think this behavior is too confusing, we can just remove it.

As a side note, I thought [you were in favor of this](https://github.com/redis-rs/redis-rs/issues/1043#issuecomment-3053383125) :)

---

done